### PR TITLE
DCS-1373 Updating dependencies

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: audit-ci.json [hunk 1]
```diff
@@ -2,5 +2,5 @@
   "moderate": true,
   "package-manager": "auto",
   "registry": "https://registry.npmjs.org",
-  "allowlist": [1002401]
+  "allowlist": [1004946]
```
Comment (andrewrlee) [inline audit-ci.json:5]: old one no longer relevant, new one ansi-regex, only affecting dev dependencies

File: mock-server/routes/prisoners.js [hunk 2]
```diff
@@ -72,9 +72,12 @@ router.get('/', (req, res) => {
       }
       if (lastName) {
         if (!firstName) {
-          return lastName.toUpperCase() === prisoner.lastName.toUpperCase()
+          return lastName.toString().toUpperCase() === prisoner.lastName.toUpperCase()
```
Comment (andrewrlee) [inline mock-server/routes/prisoners.js:75]: converting query param through to string

File: server/app.js [hunk 3]
```diff
@@ -227,7 +225,7 @@ module.exports = function createApp({
   // Don't cache dynamic resources
   app.use(noCache())
 
-  const excludedPaths = new RegExp('^/public|^/assets|^/health|^/ping|^/favicon.ico')
+  const excludedPaths = /^\/public|^\/assets|^\/health|^\/ping|^\/favicon.ico/
```
Comment (andrewrlee) [inline server/app.js:228]: new eslint rule

File: server/data/audit.js [hunk 4]
```diff
@@ -45,7 +45,7 @@ module.exports = {
       })
   },
 
-  async getEvents(action, filters = {}, startMoment, endMoment) {
+  async getEvents(action, filters, startMoment, endMoment) {
```
Comment (andrewrlee) [inline server/data/audit.js:48]: new eslint rule doesn't like default args not being the last one - this is only called in one place and always passes through a value

File: server/routes/admin/licence.ts [hunk 5]
```diff
@@ -112,7 +112,7 @@ export = (licenceService, signInService, prisonerService, audit, roNotificationH
     asyncMiddleware(async (req, res) => {
       const { abookingId: bookingId } = req.params
       const systemToken = await signInService.getClientCredentialsTokens(req.user.username)
-      const [ignored, error] = unwrapResult(
+      const [, error] = unwrapResult(
```
Comment (andrewrlee) [inline server/routes/admin/licence.ts:115]: new eslint rule not liking unused variable

File: server/routes/api.js [hunk 6]
```diff
@@ -19,8 +19,8 @@ module.exports = ({ reportingService }) => {
     const { report } = req.params
     const { start, end } = req.query
 
-    const startDate = start ? moment(start, 'DD-MM-YYYY') : null
-    const endDate = end ? moment(end, 'DD-MM-YYYY').set({ hour: 23, minute: 59 }) : null
+    const startDate = start ? moment(start.toString(), 'DD-MM-YYYY') : null
+    const endDate = end ? moment(end.toString(), 'DD-MM-YYYY').set({ hour: 23, minute: 59 }) : null
```
Comment (andrewrlee) [inline server/routes/api.js:23]: converting query param through to string

File: server/services/licenceService.ts [hunk 7]
```diff
@@ -74,7 +74,7 @@ export class LicenceService {
 
   async getLicence(bookingId: number): Promise<LicenceRecord> {
     try {
-      const [rawLicence, rawVersionDetails] = await Promise.all<CaseWithVaryVersion, ApprovedLicenceVersion>([
+      const [rawLicence, rawVersionDetails] = await Promise.all([
```
Comment (andrewrlee) [inline server/services/licenceService.ts:77]: new type inference capability in typescript means this is unnecessary

File: test/routes/contact.test.ts [hunk 8]
```diff
@@ -19,7 +18,7 @@ describe('/contact', () => {
 
   beforeEach(() => {
     roService = new RoService(undefined, undefined)
-    mocked(roService).findResponsibleOfficer.mockResolvedValue({
+    ;(jest as any).mocked(roService).findResponsibleOfficer.mockResolvedValue({
```
Comment (andrewrlee) [inline test/routes/contact.test.ts:21]: this is gnarly - need to wait until jest type defs are updated to match latest release

File: test/supertestSetup.js [hunk 9]
```diff
@@ -109,8 +109,10 @@ const appSetup = (route, user = 'caUser', prefix = '', flash = jest.fn().mockRet
     res.locals.user = userObj
     next()
   })
-  app.use(passport.initialize())
+
   app.use(cookieSession({ keys: [''] }))
+  app.use(passport.initialize())
```
Comment (andrewrlee) [inline test/supertestSetup.js:114]: passport requires both these or get `req.isAuthenticated() not a function` type error

File: types/express.d.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
+declare namespace Express {
```
Comment (andrewrlee) [inline types/express.d.ts:1]: now using express types so making customisations available

File: server/routes/admin/licence.ts (hunk 5 above)
Comment (GurnankCheema) [reply server/routes/admin/licence.ts:115]: then still needs comma?

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Pull request merged**